File: solutions/clojure/armstrong-numbers/1/src/armstrong_numbers.clj
(ns armstrong-numbers)

(defn get-digits
  [num]
  (if (= 0 num) '() (cons (mod num 10) (get-digits (bigint (/ num 10)))))
  )

(defn pow [base exp] (reduce * (repeat exp base)))

(defn armstrong?
  "Returns true if the given number is an Armstrong number; otherwise, returns false"
  [num]
  (let [digits (get-digits num)]
    (= num (reduce + (map (fn [d] (pow d (count digits))) digits)))
    )
  )

File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
def is_armstrong_number(number):
    digits = get_digits(number)
    return number == sum(d ** len(digits) for d in digits)

def get_digits(number):
    if number == 0: 
        return []
    else: 
        return [number % 10, *get_digits(number // 10)]

